[TOC]

# å‰æƒ…æè¦

- [NBTï¼šQIIME 2å¯é‡å¤ã€äº¤äº’å¼çš„å¾®ç”Ÿç‰©ç»„åˆ†æå¹³å°](https://mp.weixin.qq.com/s/-_FHxF1XUBNF4qMV1HLPkg)
- [1ç®€ä»‹å’Œå®‰è£…Introduction&Install](https://mp.weixin.qq.com/s/vlc2uIaWnPSMhPBeQtPR4w)
- [2æ’ä»¶å·¥ä½œæµç¨‹æ¦‚è¿°Workflow](https://mp.weixin.qq.com/s/qXlx1a8OQN9Ar7HYIC3OqQ)
- [3è€å¸æœºä¸Šè·¯æŒ‡å—Experienced](https://mp.weixin.qq.com/s/gJZCRzenCplCiOsDRHLhjw)
- [4äººä½“å„éƒ¨ä½å¾®ç”Ÿç‰©ç»„åˆ†æMoving Pictures](https://mp.weixin.qq.com/s/c8ZQegtfNBHZRVjjn5Gyrw)
- [Genome Biologyï¼šäººä½“å„éƒ¨ä½å¾®ç”Ÿç‰©ç»„æ—¶é—´åºåˆ—åˆ†æ](https://mp.weixin.qq.com/s/DhecHNqv4UjYpVEu48oXAw)
- [5ç²ªèŒç§»æ¤åˆ†æç»ƒä¹ FMT](https://mp.weixin.qq.com/s/cqzpLOprpClaib1FvH7bjg)
- [Microbiomeï¼šç²ªèŒç§»æ¤æ”¹å–„è‡ªé—­ç—‡](https://mp.weixin.qq.com/s/PHpg0y6_mydtCXYUwZa2Yg)
- [6æ²™æ¼ åœŸå£¤åˆ†æAtacama soil](https://mp.weixin.qq.com/s/tmXAjkl7oW3X4uagLOJu2A)
- [mSystemsï¼šå¹²æ—±å¯¹åœŸå£¤å¾®ç”Ÿç‰©ç»„çš„å½±å“](https://mp.weixin.qq.com/s/3tF6_CfSKBbtLQU4G3NpEQ)
- [7å¸•é‡‘æ£®å°é¼ æ•™ç¨‹Parkinson's Mouse](https://mp.weixin.qq.com/s/cN1sfcWFME7S4OJy4VIREg)
- [Cellï¼šè‚ é“èŒç¾¤ä¿ƒè¿›å¸•é‡‘æ£®å‘ç”ŸParkinsonDisease](https://mp.weixin.qq.com/s/OINhALYIaH-JZICpU68icQ)
- [8å·®å¼‚ä¸°åº¦åˆ†ægneiss](https://mp.weixin.qq.com/s/wx9dr5e2B_YyqTdPJ7dVsQ)
- [9æ•°æ®å¯¼å…¥Importing data](https://mp.weixin.qq.com/s/u0k38x4lAUaghua2FDD1mQ)
- [10æ•°æ®å¯¼å‡ºExporting data](https://mp.weixin.qq.com/s/pDxDsm8vabpe9KtcLRYWxg)
- [11å…ƒæ•°æ®Metadata](https://mp.weixin.qq.com/s/Q-YTeXH84lgBbRwuzc1bsg)
- [12æ•°æ®ç­›é€‰Filtering data](https://mp.weixin.qq.com/s/zk-pXJs4GNwb1AOBPzCaHA)
- [13è®­ç»ƒç‰¹å¾åˆ†ç±»å™¨Training feature classifiers](https://mp.weixin.qq.com/s/jTRUYgacH5WszsHJVbbh4g)
- [14æ•°æ®è¯„ä¼°å’Œè´¨æ§Evaluating and controlling](https://mp.weixin.qq.com/s/1b3Hj23bKWfTkHKAPNmCBQ)
- [15æ ·å“åˆ†ç±»å’Œå›å½’q2-sample-classifier](https://mp.weixin.qq.com/s/3DGvuD3R9atSoo2CSrUJBw)
- [16çºµå‘å’Œæˆå¯¹æ ·æœ¬æ¯”è¾ƒq2-longitudinal](https://mp.weixin.qq.com/s/RhRXoGqVuLumxvbba7GgSg)
- [17é‰´å®šå’Œè¿‡æ»¤åµŒåˆä½“åºåˆ—q2-vsearch](https://mp.weixin.qq.com/s/4FvR7qGTfVFdSkdKtR6_LQ)
- [18åºåˆ—åŒç«¯åˆå¹¶read-joining](https://mp.weixin.qq.com/s/xjl41rAlqMwyZSoPBX6Tww)
- [19ä½¿ç”¨q2-vsearchèšç±»OTUs](https://mp.weixin.qq.com/s/LewtnbZlJNHw9M8Bakuyeg)
- [20å®ç”¨ç¨‹åºUtilities](https://mp.weixin.qq.com/s/kSuK4njtfGN_Ph9rmF88Lw)

# ä½¿ç”¨q2-phylogenyè¿›è¡Œç³»ç»Ÿå‘è‚²æ¨æ–­

**Phylogenetic inference with q2-phylogeny**

https://docs.qiime2.org/2020.2/tutorials/phylogeny/

å¼€å§‹æœ¬èŠ‚åˆ†æå‰ï¼Œæˆ‘ä»¬å‡å®šä½ å·²ç»å®Œæˆäº†å‰é¢å†…å®¹çš„å­¦ä¹ ã€‚

åœ¨QIIME 2ä¸­å¯ç”¨çš„å‡ ä¸ªä¸‹æ¸¸å¤šæ ·æ€§æŒ‡æ ‡è¦æ±‚ä½¿ç”¨æ­£åœ¨ç ”ç©¶çš„æ“ä½œåˆ†ç±»å•ä½ï¼ˆOTUï¼‰æˆ–ç²¾ç¡®åºåˆ—å˜ä½“ï¼ˆESVï¼Œä¹Ÿå«æ‰©å¢åºåˆ—å˜ä½“ASVï¼‰æ¥æ„å»ºç³»ç»Ÿæ ‘ã€‚

ä½†æ˜¯ï¼Œæˆ‘ä»¬å¦‚ä½•ä»åºåˆ—æ•°æ®ä¸­æ„å»ºç³»ç»Ÿå‘è‚²ï¼Ÿ

å¥½äº†ï¼Œæˆ‘ä»¬å¯ä»¥ä½¿ç”¨ä¸¤ç§åŸºäºç³»ç»Ÿå‘è‚²çš„æ–¹æ³•ã€‚ä½¿ç”¨å“ªç§å†³å®šï¼Œå¾ˆå¤§ç¨‹åº¦ä¸Šå–å†³äºæ‚¨çš„ç§‘å­¦é—®é¢˜ï¼š

1. [åŸºäºå‚è€ƒçš„ç‰‡æ®µæ’å…¥æ–¹æ³•(A reference-based fragment insertion approach)](https://doi.org/10.1128/mSystems.00021-18)ã€‚è¿™å¯èƒ½æ˜¯ç†æƒ³çš„é€‰æ‹©ã€‚ç‰¹åˆ«æ˜¯ï¼Œå¦‚æœæ‚¨çš„å‚è€ƒç³»ç»Ÿå‘è‚²å²ï¼ˆå’Œç›¸å…³çš„ä»£è¡¨æ€§åºåˆ—ï¼‰åŒ…å«æ‚¨çš„åºåˆ—å¯ä»¥è¢«å¯é åœ°æ’å…¥é‚»è¿‘çš„äº²æˆšã€‚ä¸ä¼šæ’å…¥ä¸å‚è€ƒçš„åŒ¹é…ç¨‹åº¦ä¸é«˜çš„ä»»ä½•åºåˆ—ã€‚ä¾‹å¦‚ï¼Œå¦‚æœæ‚¨çš„æ•°æ®åŒ…å«åœ¨å‚è€ƒç³»ç»Ÿå‘è‚²ä¸­æœªå¾ˆå¥½è¡¨ç¤ºçš„åºåˆ—ï¼ˆä¾‹å¦‚ä¸¢å¤±çš„è¿›åŒ–æç­‰ï¼‰ï¼Œåˆ™æ­¤æ–¹æ³•å¯èƒ½æ— æ³•å¾ˆå¥½åœ°å·¥ä½œã€‚æœ‰å…³æ›´å¤šä¿¡æ¯ï¼Œè¯·æŸ¥çœ‹è¿™äº›[å‡ºè‰²çš„ç‰‡æ®µæ’å…¥ç¤ºä¾‹](https://library.qiime2.org/plugins/q2-fragment-insertion/16/)ã€‚

2. ä»å¤´å¼€å§‹ã€‚å¯ä»¥åœ¨ä¸åŒåˆ†ç±»å•å…ƒä¸­å…¨å±€æ¯”å¯¹çš„æ ‡è®°åŸºå› é€šå¸¸é€‚åˆé€šè¿‡è¿™ç§æ–¹æ³•è¿›è¡Œåºåˆ—æ¯”å¯¹å’Œç³»ç»Ÿå‘è‚²ç ”ç©¶ã€‚åœ¨æ„å»ºä»å¤´ç³»ç»Ÿå‘è‚²å²æ—¶ï¼Œè¯·æ³¨æ„åºåˆ—çš„é•¿åº¦ï¼ŒçŸ­è¯»å¯èƒ½ä¼šå¯¼è‡´ç¼ºä¹è¶³å¤Ÿçš„ç³»ç»Ÿå‘è‚²ä¿¡æ¯ä»¥æ•è·æœ‰æ„ä¹‰çš„ç³»ç»Ÿå‘è‚²ã€‚æœ¬ç¤¾åŒºæ•™ç¨‹å°†é‡ç‚¹ä»‹ç»ä»å¤´æ–¹æ³•ã€‚

åœ¨è¿™é‡Œï¼Œæ‚¨å°†å­¦ä¹ å¦‚ä½•ä½¿ç”¨ä»å¤´ç³»ç»Ÿå‘è‚²æ–¹æ³•æ¥ï¼š

1. åœ¨QIIME 2ä¸­ç”Ÿæˆå¤šåºåˆ—å¯¹é½æ–‡ä»¶
2. å¦‚æœéœ€è¦ï¼Œé®ç½©å¯¹é½ä¸­çš„éä¿å®ˆåŒº
3. æ„é€ ç³»ç»Ÿå‘è‚²æ ‘
4. ç¡®å®šç³»ç»Ÿå‘è‚²æ ‘çš„æ ¹

å¦‚æœæ‚¨æƒ³é€šè¿‡ä½¿ç”¨QIIME 2å¤–éƒ¨çš„å·¥å…·æ¥æ›¿ä»£æ­¤å¤„æ¦‚è¿°çš„ä»»ä½•æ­¥éª¤ï¼Œè¯·åœ¨é€‚å½“çš„åœ°æ–¹æŸ¥çœ‹[9æ•°æ®å¯¼å…¥Importing data](https://mp.weixin.qq.com/s/u0k38x4lAUaghua2FDD1mQ)ï¼Œ[10æ•°æ®å¯¼å‡ºExporting data](https://mp.weixin.qq.com/s/pDxDsm8vabpe9KtcLRYWxg)å’Œ[12æ•°æ®ç­›é€‰Filtering data](https://mp.weixin.qq.com/s/zk-pXJs4GNwb1AOBPzCaHA)æ–‡æ¡£ã€‚

## åºåˆ—å¯¹é½Sequence Alignment

åœ¨æ„å»º[ç³»ç»Ÿå‘è‚²](https://simple.wikipedia.org/wiki/Phylogeny)ä¹‹å‰ï¼Œæˆ‘ä»¬å¿…é¡»ç”Ÿæˆ[å¤šåºåˆ—æ¯”å¯¹ï¼ˆmultiple sequence alignmentï¼ŒMSAï¼‰](https://en.wikipedia.org/wiki/Multiple_sequence_alignment)ã€‚ åœ¨æ„å»ºMSAæ—¶ï¼Œæˆ‘ä»¬å°†é€šè¿‡å…¶[åºåˆ—ç›¸ä¼¼æ€§](https://doi.org/10.1002/0471250953.bi0301s42)æ¥è¯´æ˜æ¯”å¯¹æ®‹åŸºï¼ˆMSAåˆ—ï¼‰çš„[æ¨å®šåŒæºæ€§](https://doi.org/10.1006/mpev.2000.0785)ã€‚

æ„å»ºMSAçš„ç®—æ³•æ•°é‡ä¼—å¤šã€‚ æˆ‘ä»¬å°†é€šè¿‡[q2-alignment](https://docs.qiime2.org/2020.2/plugins/available/alignment/)æ’ä»¶ä½¿ç”¨[MAFFTï¼ˆä½¿ç”¨å¿«é€Ÿå‚…é‡Œå¶å˜æ¢çš„å¤šé‡å¯¹é½ï¼‰](https://en.wikipedia.org/wiki/MAFFT)ã€‚ æœ‰å…³æ›´å¤šä¿¡æ¯ï¼Œè¯·æŸ¥çœ‹[MAFFTçš„æ–‡ç« ](https://doi.org/10.1093/molbev/mst010)ã€‚

é¦–å…ˆï¼Œåˆ›å»ºä¸€ä¸ªå·¥ä½œç›®å½•ï¼š

    mkdir -p qiime2-phylogeny-tutorial
    cd qiime2-phylogeny-tutorial

ä¸‹è½½ä»£è¡¨åºåˆ— 

    wget -c "https://data.qiime2.org/2020.2/tutorials/phylogeny/rep-seqs.qza" \
      -O "rep-seqs.qza"

è¿è¡ŒMAFFT

    # 6s
    time qiime alignment mafft \
      --i-sequences rep-seqs.qza \
      --o-alignment aligned-rep-seqs.qza

è¾“å‡ºå¯¹è±¡ï¼š
- `rep-seqs.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Frep-seqs.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/rep-seqs.qza)
- `aligned-rep-seqs.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Faligned-rep-seqs.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/aligned-rep-seqs.qza)


## å‡å°‘å¯¹é½çš„æ­§ä¹‰ï¼šå±è”½å’Œå‚è€ƒå¯¹é½æ–¹å¼ Reducing alignment ambiguity: masking and reference alignments

ä¸ºä»€ä¹ˆè¦å±è”½(mask)åºåˆ—ï¼Ÿ

å±è”½/æ©ç æœ‰åŠ©äºæ¶ˆé™¤åœ¨ç³»ç»Ÿå‘è‚²åˆ†æä¹‹å‰ç³»ç»Ÿå‘è‚²ä¸Šæ— ç”¨æˆ–è¯¯å¯¼çš„æ¯”å¯¹åˆ—ã€‚ è®¸å¤šæ—¶é—´å¯¹å‡†è¯¯å·®ä¼šå¼•å…¥å™ªå£°å¹¶æ··æ·†ç³»ç»Ÿå‘è‚²æ¨æ–­ã€‚ åœ¨æ‰§è¡Œç³»ç»Ÿå‘ç”Ÿæ¨æ–­ä¹‹å‰ï¼Œé€šå¸¸ä¼šå±è”½ï¼ˆåˆ é™¤ï¼‰è¿™äº›æ­§ä¹‰å¯¹é½çš„åŒºåŸŸã€‚ ç‰¹åˆ«æ˜¯ï¼ŒDavid Laneï¼ˆ1991ï¼‰çš„[16S / 23S rRNAæµ‹åº](http://catdir.loc.gov/catdir/toc/onix05/90012998.html)å»ºè®®åœ¨ç³»ç»Ÿå‘è‚²åˆ†æä¹‹å‰å±è”½SSUæ•°æ®ã€‚ ä½†æ˜¯ï¼ŒçŸ¥é“å¦‚ä½•å¤„ç†æ­§ä¹‰å¯¹é½çš„åŒºåŸŸä»¥åŠä½•æ—¶ä½¿ç”¨å±è”½å¾ˆå¤§ç¨‹åº¦ä¸Šå–å†³äºè¦åˆ†æçš„æ ‡å¿—ç‰©åŸºå› å’Œæ•°æ®éœ€è¦å›ç­”çš„ç§‘å­¦é—®é¢˜ã€‚

> æ³¨æ„
> 
> è¯·è®°ä½ï¼Œè¿™ä»ç„¶æ˜¯ä¸€ä¸ªæ´»è·ƒçš„è®¨è®ºé¢†åŸŸï¼Œä»¥ä¸‹éè¯¦å°½çš„æ–‡ç« æ¸…å•çªæ˜¾äº†è¿™ä¸€ç‚¹ï¼š[Wuç­‰2012å¹´](https://doi.org/10.1371/journal.pone.0030288)ï¼Œ[Ashkenazyç­‰äºº2018å¹´](https://doi.org/10.1093/sysbio/syy036)ï¼Œ[Schloss 2010å¹´](https://doi.org/10.1371/journal.pcbi.1000844)ï¼Œ[Tanç­‰äºº2015å¹´](https://doi.org/10.1093/sysbio/syv033)ï¼Œ[Rajan2015å¹´](https://doi.org/10.1093/sysbio/syv033)ã€‚

å¦‚ä½•å±è”½åºåˆ—å‘¢ï¼Ÿ

ä¸ºäº†æˆ‘ä»¬çš„ç›®çš„ï¼Œæˆ‘ä»¬å‡è®¾åœ¨ä¸Šé¢ç”Ÿæˆçš„MAFFTå¯¹é½æ–¹å¼ä¸­å­˜åœ¨ä¸ç¡®å®šçš„å¯¹é½åˆ—ã€‚ å±è”½å¯¹é½é»˜è®¤è®¾ç½®`--p-min-conservation`æ¥è¿‘QIIME 1çš„Lane maskã€‚è¯·å¯†åˆ‡æ³¨æ„å¯¹é½æ’ä»¶çš„æ›´æ–°ã€‚

    # 6s
    time qiime alignment mask \
      --i-alignment aligned-rep-seqs.qza \
      --o-masked-alignment masked-aligned-rep-seqs.qza
  
è¾“å‡ºå¯¹è±¡ï¼š
- `masked-aligned-rep-seqs.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmasked-aligned-rep-seqs.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/masked-aligned-rep-seqs.qza)

åŸºäºå‚è€ƒçš„å¤šåºåˆ—æ¯”å¯¹(Reference based alignments)

æœ‰å¤šç§å·¥å…·ï¼Œä¾‹å¦‚[PyNAST](https://doi.org/10.1093/bioinformatics/btp636)ï¼ˆä½¿ç”¨[NAST](https://doi.org/10.1093/nar/gkl244)ï¼‰ï¼Œ[Infernal](https://doi.org/10.1093/bioinformatics/btt509)å’Œ[SINA](https://doi.org/10.1093/bioinformatics/bts252)ç­‰ï¼Œå®ƒä»¬è¯•å›¾é€šè¿‡ä½¿ç”¨æ ¡å¯¹çš„å‚è€ƒæ¯”å¯¹ï¼ˆä¾‹å¦‚[SILVA](https://www.arb-silva.de/)ï¼‰æ¥å‡å°‘æ­§ä¹‰æ¯”å¯¹åŒºåŸŸçš„æ•°é‡ã€‚å‚è€ƒæ¯”å¯¹å¯¹rRNAåŸºå› ç‰¹åˆ«æœ‰æ•ˆåºåˆ—æ•°æ®ï¼Œå› ä¸ºå°†äºŒçº§ç»“æ„çš„çŸ¥è¯†æ•´åˆåˆ°äº†æ ¡å¯¹è¿‡ç¨‹ä¸­ï¼Œä»è€Œæé«˜äº†æ¯”å¯¹è´¨é‡ã€‚æœ‰å…³åŸºäºå‚è€ƒçš„æ¯”å¯¹æ–¹æ³•æ›´æ·±å…¥ï¼Œæ›´æœ‰è¯´æœåŠ›çš„æ¦‚è¿°ï¼Œè¯·æŸ¥çœ‹[SINAç¤¾åŒºæ•™ç¨‹](https://forum.qiime2.org/t/q2-alignment-reference-based-alignment-using-sina/6220)ã€‚

> æ³¨æ„
> 
> å°±åƒä¸Šé¢çš„MAFFTç¤ºä¾‹ä¸€æ ·ï¼Œä½¿ç”¨åŸºäºå‚è€ƒçš„å¯¹é½æ–¹å¼æ„é€ çš„å¯¹é½æ–¹å¼ä¹Ÿå¯ä»¥è¢«å±è”½ã€‚ åŒæ ·ï¼Œæˆ‘ä»¬åœ¨æ­¤å¤„è®¨è®ºçš„å‚è€ƒå¯¹é½æ–¹æ³•ä¸æˆ‘ä»¬ä¹‹å‰æåˆ°çš„å‚è€ƒç³»ç»Ÿå‘ç”Ÿæ–¹æ³•ï¼ˆå³[q2-fragment-insertion](https://docs.qiime2.org/2020.2/plugins/available/fragment-insertion/)ï¼‰ä¸åŒã€‚ ä¹Ÿå°±æ˜¯è¯´ï¼Œæˆ‘ä»¬ä¸æ˜¯å°†æ•°æ®æ’å…¥åˆ°ç°æœ‰çš„æ ‘ä¸­ï¼Œè€Œåªæ˜¯è¯•å›¾åˆ›å»ºæ›´å¥å£®çš„æ¯”å¯¹ï¼Œä»¥å®ç°æ›´å¥½çš„ä»å¤´ç³»ç»Ÿå‘è‚²ã€‚

## æ„å»ºç³»ç»Ÿå‘è‚² Construct a phylogeny

ä¸MSAç®—æ³•ä¸€æ ·ï¼Œç³»ç»Ÿå‘è‚²æ¨æ–­å·¥å…·ä¹Ÿå¾ˆä¸°å¯Œã€‚ å¹¸è¿çš„æ˜¯ï¼Œæœ‰å¾ˆå¤šå¾ˆæ£’çš„èµ„æºå¯ä»¥å­¦ä¹ ç³»ç»Ÿå‘è‚²å­¦ã€‚ ä»¥ä¸‹æ˜¯ä¸€äº›å…¥é—¨èµ„æºï¼Œå¯å¸®åŠ©æ‚¨å…¥é—¨ï¼š

1. [ç³»ç»Ÿå‘è‚²å­¦å…¥é—¨æ•™ç¨‹](https://doi.org/10.1016/S0168-9525(03)00112-4)
2. [åˆ†å­ç³»ç»Ÿå‘è‚²å­¦-åŸç†ä¸å®è·µ](https://dx.doi.org/10.1038/nrg3186)
3. [ç³»ç»Ÿå‘è‚²å­¦-ç®€ä»‹](https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics)

qiime2ï¼šçš„ `q2-phylogeny` æ’ä»¶ä¸­æä¾›äº†å‡ ç§æ–¹æ³•/æµç¨‹ã€‚ è¿™äº›åŸºäºä»¥ä¸‹å·¥å…·ï¼š

1. [FastTree](https://doi.org/10.1371/journal.pone.0009490)
2. [RAxML](https://doi.org/10.1093/bioinformatics/btu033)
3. [IQ-TREE](https://doi.org/10.1093/molbev/msu300)

# æ–¹æ³•Methods

## fasttree

FastTreeèƒ½å¤Ÿä»å¤§å‹åºåˆ—æ¯”å¯¹ä¸­å¿«é€Ÿæ„å»ºç³»ç»Ÿå‘è‚²ã€‚ å®ƒé€šè¿‡ä½¿ç”¨[CAT-like](https://doi.org/10.1109/IPDPS.2006.1639535)çš„é€Ÿç‡ç±»åˆ«è¿‘ä¼¼å€¼æ¥å®ç°æ­¤ç›®çš„ï¼Œè¯¥è¿‘ä¼¼å€¼ä¹Ÿå¯ä»¥é€šè¿‡RAxMLè·å¾—ï¼ˆå¦‚ä¸‹æ‰€è¿°ï¼‰ã€‚ æŸ¥é˜…[FastTreeåœ¨çº¿æ‰‹å†Œ](http://www.microbesonline.org/fasttree/)ä»¥è·å–æ›´å¤šä¿¡æ¯ã€‚

    # 6s
    time qiime phylogeny fasttree \
      --i-alignment masked-aligned-rep-seqs.qza \
      --o-tree fasttree-tree.qza

è¾“å‡ºå¯¹è±¡ï¼š
- `fasttree-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Ffasttree-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/fasttree-tree.qza)

> å°è´´ç¤ºï¼šä¸ºäº†æ–¹ä¾¿è€Œç›´æ¥åœ°æŸ¥çœ‹æ‚¨çš„`tree.qza`æ–‡ä»¶ï¼Œè¯·å°†å®ƒä»¬ä¸Šä¼ åˆ°[iTOL](https://itol.embl.de/)ã€‚ åœ¨è¿™é‡Œï¼Œæ‚¨å¯ä»¥äº¤äº’åœ°æŸ¥çœ‹å’Œæ“çºµç³»ç»Ÿå‘è‚²æ ‘ã€‚ æ›´å¥½çš„æ˜¯ï¼Œåœ¨â€œæ­£å¸¸æ¨¡å¼â€ä¸‹æŸ¥çœ‹æ ‘æ‹“æ‰‘æ—¶ï¼Œæ‚¨å¯ä»¥å°†å…³è”çš„`alignment.qza`ï¼ˆç”¨äºæ„å»ºç³»ç»Ÿå‘è‚²çš„é‚£ä¸ªï¼‰æˆ–ç›¸å…³çš„`taxonomy.qza`æ–‡ä»¶æ‹–æ”¾åˆ°iTOLæ ‘å¯è§†åŒ–ä¸­ã€‚ è¿™å°†å…è®¸æ‚¨ç›´æ¥æŸ¥çœ‹åºåˆ—æ¯”å¯¹æˆ–åˆ†ç±»å­¦ä»¥åŠç³»ç»Ÿå‘è‚²ã€‚ ğŸ•¶ï¸

> è¯‘è€…æµ‹è¯•itolå¯¹QIIME 2çš„qzaæ ¼å¼çš„æ”¯æŒæƒ…å†µå’Œå±•ç¤ºæ•ˆæœã€‚

> è®¿é—®ttps://itol.embl.de/ï¼Œæ³¨å†Œè´¦å·ç™»é™†ã€‚ä¸Šä¼ æ ‘æ–‡ä»¶`fasttree-tree.qza`

![image](http://210.75.224.110/github/QIIME2ChineseManual/fig/2020.2/phylogeny.itol1.jpg)

> æ‹–æ‹½`aligned-rep-seqs.qza`è‡³æ ‘ä¸Šå±•ç¤º

![image](http://210.75.224.110/github/QIIME2ChineseManual/fig/2020.2/phylogeny.itol2.jpg)

æ›´å¤šitolçš„ä½¿ç”¨ï¼Œè¯¦è§ - [iTOLç¾åŒ–](https://mp.weixin.qq.com/s/htW2nWJBCp555y27grrGRw) [è¿›é˜¶](https://mp.weixin.qq.com/s/ZATPWf47WBKl91gFEBzusg)  

## raxml

åƒfasttreeä¸€æ ·ï¼Œraxmlå°†æ‰§è¡Œå•ä¸ªç³»ç»Ÿå‘è‚²æ¨æ–­å¹¶è¿”å›ä¸€æ£µæ ‘ã€‚ æ³¨æ„ï¼Œraxmlçš„é»˜è®¤æ¨¡å‹æ˜¯`--p-substitution-model GTRGAMMA`ã€‚ å¦‚æœæ‚¨æƒ³ä½¿ç”¨åƒfasttreeè¿™æ ·çš„CATæ¨¡å‹æ¥æ„å»ºæ ‘ï¼Œåªéœ€å°†GTRGAMMAæ›¿æ¢ä¸ºGTRCATï¼Œå¦‚ä¸‹æ‰€ç¤ºï¼š

    # 6s
    time qiime phylogeny raxml \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-substitution-model GTRCAT \
      --o-tree raxml-cat-tree.qza \
      --verbose

æ­£å¸¸æ˜¯æ˜¾ç¤ºå¦‚ä¸‹ç»“æœï¼š

    Running external command line application. This may print messages to stdout and/or stderr.
    The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.
    
    Command: raxmlHPC -m GTRCAT -p 6837 -N 1 -s /tmp/qiime2-archive-f7qd3egy/0f292022-14c0-431e-83ea-87e07fa04b35/data/aligned-dna-sequences.fasta -w /tmp/tmpscvv747b -n q2
    
    Warning, you specified a working directory via "-w"
    Keep in mind that RAxML only accepts absolute path names, not relative ones!
    
    RAxML can't, parse the alignment file as phylip file 
    it will now try to parse it as FASTA file
    
    
    
    Using BFGS method to optimize GTR rate parameters, to disable this specify "--no-bfgs" 
    
    
    
    This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.
    
    With greatly appreciated code contributions by:
    Andre Aberer      (HITS)
    Simon Berger      (HITS)
    Alexey Kozlov     (HITS)
    Kassian Kobert    (HITS)
    David Dao         (KIT and HITS)
    Sarah Lutteropp   (KIT and HITS)
    Nick Pattengale   (Sandia)
    Wayne Pfeiffer    (SDSC)
    Akifumi S. Tanabe (NRIFS)
    Charlie Taylor    (UF)
    
    
    Alignment has 157 distinct alignment patterns
    
    Proportion of gaps and completely undetermined characters in this alignment: 39.77%
    
    RAxML rapid hill-climbing mode
    
    Using 1 distinct models/data partitions with joint branch length optimization
    
    
    Executing 1 inferences on the original alignment using 1 distinct randomized MP trees
    
    All free model parameters will be estimated by RAxML
    ML estimate of 25 per site rate categories
    
    Likelihood of final tree will be evaluated and optimized under GAMMA
    
    GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units
    
    Partition: 0
    Alignment Patterns: 157
    Name: No Name Provided
    DataType: DNA
    Substitution Matrix: GTR
    
    
    
    
    RAxML was called as follows:
    
    raxmlHPC -m GTRCAT -p 6837 -N 1 -s /tmp/qiime2-archive-f7qd3egy/0f292022-14c0-431e-83ea-87e07fa04b35/data/aligned-dna-sequences.fasta -w /tmp/tmpscvv747b -n q2 
    
    
    Partition: 0 with name: No Name Provided
    Base frequencies: 0.243 0.182 0.319 0.256 
    
    Inference[0]: Time 1.304490 CAT-based likelihood -1246.372627, best rearrangement setting 5
    
    
    Conducting final model optimizations on all 1 trees under GAMMA-based models ....
    
    Inference[0] final GAMMA-based Likelihood: -1393.280071 tree written to file /tmp/tmpscvv747b/RAxML_result.q2
    
    
    Starting final GAMMA-based thorough Optimization on tree 0 likelihood -1393.280071 .... 
    
    Final GAMMA-based Score of best tree -1387.733722
    
    Program execution info written to /tmp/tmpscvv747b/RAxML_info.q2
    Best-scoring ML tree written to: /tmp/tmpscvv747b/RAxML_bestTree.q2
    
    Overall execution time: 2.139081 secs or 0.000594 hours or 0.000025 days
    
    Saved Phylogeny[Unrooted] to: raxml-cat-tree.qza

**è¾“å‡ºå¯¹è±¡ï¼š**
- `raxml-cat-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/raxml-cat-tree.qza)

è¯¦è€…æ³¨ï¼šå¦‚æœåˆ†æä¸­æŠ¥é”™ï¼Œå¯èƒ½æ˜¯ç¯å¢ƒå˜é‡æ±¡æŸ“ï¼Œå¯ä»¥é€€å‡ºç¯å¢ƒã€‚é‡æ–°ç™»é™†termailï¼Œå†åŠ è½½qiime2ç¯å¢ƒé‡è¯•ã€‚

### ä½¿ç”¨raxmlæ‰§è¡Œå¤šæ¬¡æœç´¢ Perform multiple searches using raxml

å¦‚æœæ‚¨æƒ³å¯¹â€œæ ‘ç©ºé—´(tree space)â€æ‰§è¡Œæ›´å½»åº•çš„æœç´¢ï¼Œåˆ™å¯ä»¥ä½¿ç”¨`--pn-searches 5`æŒ‡ç¤ºraxmlå¯¹å®Œå…¨å¯¹é½æ‰§è¡Œå¤šä¸ªç‹¬ç«‹æœç´¢ã€‚ä¸€æ—¦å®Œæˆè¿™5ä¸ªç‹¬ç«‹æœç´¢ï¼Œåˆ™ä»…æ‰§è¡Œä¸€æ¬¡æœ€ä½³æœç´¢ å¾—åˆ†æ ‘å°†è¢«è¿”å›ã€‚ *è¯·æ³¨æ„ï¼Œæˆ‘ä»¬ä¸åœ¨è¿™é‡Œè¿›è¡Œè‡ªå±•æ”¯æŒç‡çš„è®¡ç®—ï¼Œæˆ‘ä»¬å°†åœ¨ä»¥åçš„ç¤ºä¾‹ä¸­è¿›è¡Œä»‹ç»*ã€‚ è®©æˆ‘ä»¬è®¾ç½®`--p-substitution-model GTRCAT`ã€‚ æœ€åï¼Œæˆ‘ä»¬è¿˜è¦é€šè¿‡`--p-seed`æ‰‹åŠ¨è®¾ç½®ç§å­ã€‚ é€šè¿‡è®¾ç½®æˆ‘ä»¬çš„ç§å­ï¼Œæˆ‘ä»¬å…è®¸å…¶ä»–ç”¨æˆ·å¯ä»¥é‡å¤å‡ºæˆ‘ä»¬çš„ç³»ç»Ÿå‘è‚²æ ‘ã€‚ ä¹Ÿå°±æ˜¯è¯´ï¼Œä»»ä½•ä½¿ç”¨ç›¸åŒåºåˆ—æ¯”å¯¹å’Œæ›¿æ¢æ¨¡å‹çš„äººï¼Œåªè¦è®¾ç½®äº†ç›¸åŒçš„ç§å­å€¼ï¼Œéƒ½å°†ç”Ÿæˆç›¸åŒçš„æ ‘ã€‚ å°½ç®¡`--p-seed`ä¸æ˜¯å¿…éœ€çš„å‚æ•°ï¼Œä½†æ˜¯è®¾ç½®æ­¤å€¼é€šå¸¸æ˜¯ä¸€ä¸ªå¥½ä¸»æ„ã€‚

    # 13s
    time qiime phylogeny raxml \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-substitution-model GTRCAT \
      --p-seed 1723 \
      --p-n-searches 5 \
      --o-tree raxml-cat-searches-tree.qza \
      --verbose

è¾“å‡ºå†…å®¹å¦‚ä¸‹ï¼š

    Running external command line application. This may print messages to stdout and/or stderr.
    The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.
    
    Command: raxmlHPC -m GTRCAT -p 1723 -N 5 -s /tmp/qiime2-archive-5mraa_k4/0f292022-14c0-431e-83ea-87e07fa04b35/data/aligned-dna-sequences.fasta -w /tmp/tmp9k6jpgh5 -n q2
    
    Warning, you specified a working directory via "-w"
    Keep in mind that RAxML only accepts absolute path names, not relative ones!
    
    RAxML can't, parse the alignment file as phylip file 
    it will now try to parse it as FASTA file
    
    Using BFGS method to optimize GTR rate parameters, to disable this specify "--no-bfgs" 
    
    This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.
    
    With greatly appreciated code contributions by:
    Andre Aberer      (HITS)
    ...
    Charlie Taylor    (UF)
    
    Alignment has 157 distinct alignment patterns
    
    Proportion of gaps and completely undetermined characters in this alignment: 39.77%
    
    RAxML rapid hill-climbing mode
    
    Using 1 distinct models/data partitions with joint branch length optimization
    
    Executing 5 inferences on the original alignment using 5 distinct randomized MP trees
    
    All free model parameters will be estimated by RAxML
    ML estimate of 25 per site rate categories
    
    Likelihood of final tree will be evaluated and optimized under GAMMA
    
    GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units
    
    Partition: 0
    Alignment Patterns: 157
    Name: No Name Provided
    DataType: DNA
    Substitution Matrix: GTR
    
    RAxML was called as follows:
    
    raxmlHPC -m GTRCAT -p 1723 -N 5 -s /tmp/qiime2-archive-5mraa_k4/0f292022-14c0-431e-83ea-87e07fa04b35/data/aligned-dna-sequences.fasta -w /tmp/tmp9k6jpgh5 -n q2 
    
    
    Partition: 0 with name: No Name Provided
    Base frequencies: 0.243 0.182 0.319 0.256 
    
    Inference[0]: Time 1.251481 CAT-based likelihood -1238.242991, best rearrangement setting 5
    ...
    Inference[4]: Time 0.884891 CAT-based likelihood -1261.321621, best rearrangement setting 5
    
    
    Conducting final model optimizations on all 5 trees under GAMMA-based models ....
    
    Inference[0] final GAMMA-based Likelihood: -1388.324040 tree written to file /tmp/tmp9k6jpgh5/RAxML_result.q2.RUN.0
    ...
    Inference[4] final GAMMA-based Likelihood: -1387.557824 tree written to file /tmp/tmp9k6jpgh5/RAxML_result.q2.RUN.4
    
    
    Starting final GAMMA-based thorough Optimization on tree 4 likelihood -1387.557824 .... 
    
    Final GAMMA-based Score of best tree -1387.385915
    
    Program execution info written to /tmp/tmp9k6jpgh5/RAxML_info.q2
    Best-scoring ML tree written to: /tmp/tmp9k6jpgh5/RAxML_bestTree.q2
    
    Overall execution time: 6.214823 secs or 0.001726 hours or 0.000072 days
    
    Saved Phylogeny[Unrooted] to: raxml-cat-searches-tree.qza

**è¾“å‡ºå¯¹è±¡ï¼š**
- `raxml-cat-searches-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-searches-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/raxml-cat-searches-tree.qza)

### å¿«é€Ÿè‡ªå±• raxml-rapid-bootstrap

åœ¨ç³»ç»Ÿå‘è‚²å­¦ä¸­ï¼Œä¼˜è‰¯ä½œæ³•æ˜¯æ£€æŸ¥ç³»ç»Ÿå‘è‚²ä¸­çš„[åˆ†å¼€(splits)/åˆ†ä¸ºä¸¤éƒ¨åˆ†(bipartitions)](https://en.wikipedia.org/wiki/Split_(phylogenetics))å—æ”¯æŒçš„ç¨‹åº¦ã€‚äººä»¬é€šå¸¸å¯¹è¿›åŒ–å²ä¸­å“ªäº›è¿›åŒ–æä¸å…¶ä»–è¿›åŒ–æç‰¢å›ºåœ°åˆ†ç¦»æ„Ÿå…´è¶£ã€‚ä¸€ç§æ–¹æ³•æ˜¯é€šè¿‡è‡ªå±•(bootstrapping)ï¼ˆè¯·å‚è§ä¸Šé¢ç¬¬ä¸€ä¸ªä»‹ç»æ€§é“¾æ¥çš„â€œè‡ªå±•â€éƒ¨åˆ†ï¼‰ã€‚åœ¨QIIME 2ä¸­ï¼Œæˆ‘ä»¬æä¾›äº†å¯¹RAxML[å¿«é€Ÿè‡ªå±•](https://dx.doi.org/10.1080/10635150802429642)åŠŸèƒ½ã€‚è¯¥å‘½ä»¤ä¸å‰ä¸€ä¸ªå‘½ä»¤çš„å”¯ä¸€åŒºåˆ«æ˜¯é™„åŠ æ ‡å¿—`--p-bootstrap-replicates`å’Œ`--p-rapid-bootstrap-seed`ã€‚æ‰§è¡Œ100-1000ä¸ªè‡ªå±•ç¨‹åºé‡å¤çš„ä»»ä½•ä½ç½®éƒ½æ˜¯å¾ˆå¸¸è§çš„ã€‚ `--p-rapid-bootstrap-seed`çš„å·¥ä½œåŸç†ä¸ä¸Šé¢çš„`--p-seed`å‚æ•°éå¸¸ç›¸ä¼¼ï¼Œä¸åŒä¹‹å¤„åœ¨äºï¼Œå®ƒå…è®¸ä»»ä½•äººé‡ç°è‡ªå±•è¿‡ç¨‹ä»¥åŠå¯¹åˆ†ç¦»çš„ç›¸å…³æ”¯æŒã€‚

æ ¹æ®[RAxMLåœ¨çº¿æ–‡æ¡£](https://sco.h-its.org/exelixis/web/software/raxml/hands_on.html)å’Œ[RAxMLæ‰‹å†Œ](https://sco.h-its.org/exelixis/resource/download/NewManual.pdf)ï¼Œæˆ‘ä»¬å°†åœ¨ä¸‹é¢æ‰§è¡Œçš„å¿«é€Ÿè‡ªå±•å‘½ä»¤å°†æ‰§è¡Œä»¥ä¸‹æ“ä½œï¼š

1. è‡ªå±•è¾“å…¥å¯¹é½æ–¹å¼100æ¬¡ï¼Œå¹¶å¯¹æ¯ä¸€æ¬¡æ‰§è¡Œæœ€å¤§ä¼¼ç„¶ï¼ˆMLï¼‰æœç´¢ã€‚
2. ä½¿ç”¨åŸå§‹è¾“å…¥å¯¹é½æ–¹å¼ï¼Œé€šè¿‡å¤šæ¬¡ç‹¬ç«‹æœç´¢æ‰¾åˆ°æœ€ä½³è¯„åˆ†çš„MLæ ‘ã€‚ç‹¬ç«‹æœç´¢çš„æ•°é‡ç”±ç¬¬ä¸€æ­¥ä¸­è®¾ç½®çš„å¼•å¯¼å¤åˆ¶æ•°é‡å†³å®šã€‚ä¹Ÿå°±æ˜¯è¯´ï¼Œéšç€å¼•å¯¼å¤åˆ¶çš„å¢åŠ ï¼Œæ‚¨çš„æœç´¢å°†å˜å¾—æ›´åŠ å½»åº•ã€‚
3. RAxMLçš„MLä¼˜åŒ–ä½¿ç”¨æ¯5ä¸ªè‡ªå±•æ ‘ä½œä¸ºåŸå§‹æ ‘çš„MLæœç´¢çš„èµ·å§‹æ ‘ã€‚

å°†ä¸¤éƒ¨åˆ†ï¼ˆè‡ªå±•æ”¯æŒç‡ï¼Œç¬¬ä¸€æ­¥ï¼‰æ˜ å°„åˆ°å¾—åˆ†æœ€é«˜çš„MLæ ‘ï¼ˆç¬¬äºŒæ­¥ï¼‰ä¸Šã€‚

    # 39s
    time qiime phylogeny raxml-rapid-bootstrap \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-seed 1723 \
      --p-rapid-bootstrap-seed 9384 \
      --p-bootstrap-replicates 100 \
      --p-substitution-model GTRCAT \
      --o-tree raxml-cat-bootstrap-tree.qza \
      --verbose

æ˜¾ç¤ºå¦‚ä¸‹ç»“æœï¼š

    Time for BS model parameter optimization 0.106221
    Bootstrap[0]: Time 0.340991 seconds, bootstrap likelihood -1199.758796, best rearrangement setting 12
    Bootstrap[1]: Time 0.221361 seconds, bootstrap likelihood -1344.229251, best rearrangement setting 6
    ...
    Bootstrap[99]: Time 0.108568 seconds, bootstrap likelihood -1270.157811, best rearrangement setting 7
    Saved Phylogeny[Unrooted] to: raxml-cat-bootstrap-tree.qza

**è¾“å‡ºå¯¹è±¡ï¼š**
- `raxml-cat-bootstrap-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-bootstrap-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/raxml-cat-bootstrap-tree.qza)

> å°è´´ç¤ºï¼šä¼˜åŒ–RAxMLè¿è¡Œæ—¶é—´ã€‚æ‚¨å¯èƒ½å·²ç»æ³¨æ„åˆ°ï¼Œæˆ‘ä»¬å°šæœªåœ¨RAxMLæ–¹æ³•ä¸­æ·»åŠ æ ‡å¿—`--p-raxml-version`ã€‚è¯¥å‚æ•°æä¾›äº†ä¸€ç§è®¿é—®RAxMLç‰ˆæœ¬çš„æ–¹æ³•ï¼Œè¯¥ç‰ˆæœ¬å…·æœ‰é’ˆå¯¹å„ç§ç°ä»£x86å¤„ç†å™¨ä½“ç³»ç»“æ„è¿›è¡Œäº†ä¼˜åŒ–çš„çŸ¢é‡æŒ‡ä»¤ã€‚æ‘˜è‡ªRAxMLæ‰‹å†Œå’Œå¸®åŠ©æ–‡æ¡£ï¼šé¦–å…ˆï¼Œæœ€æ–°çš„å¤„ç†å™¨å°†æ”¯æŒSSE3çŸ¢é‡æŒ‡ä»¤ï¼ˆå³å¯èƒ½æ”¯æŒæ›´å¿«çš„AVX2çŸ¢é‡æŒ‡ä»¤ï¼‰ã€‚å…¶æ¬¡ï¼Œè¿™äº›æŒ‡ä»¤å°†å¤§å¤§åŠ é€Ÿä¼¼ç„¶æ€§å’Œç®€çº¦è®¡ç®—ã€‚é€šå¸¸ï¼ŒSSE3ç‰ˆæœ¬çš„è¿è¡Œé€Ÿåº¦æ¯”æ ‡å‡†ç‰ˆæœ¬å¿«40ï¼…ã€‚ AVX2ç‰ˆæœ¬çš„è¿è¡Œé€Ÿåº¦æ¯”SSE3ç‰ˆæœ¬å¿«10-30ï¼…ã€‚æ­¤å¤–ï¼Œè¯·è®°ä½ï¼Œä½¿ç”¨æ›´å¤šçš„å†…æ ¸/çº¿ç¨‹å¹¶ä¸ä¸€å®šä¼šå‡å°‘è¿è¡Œæ—¶é—´ã€‚ RAxMLæ‰‹å†Œå»ºè®®æ¯500ä¸ªDNAæ¯”å¯¹æ¨¡å¼ä½¿ç”¨1ä¸ªæ ¸å¿ƒã€‚ä½¿ç”¨`--verbose`é€‰é¡¹æ—¶ï¼Œå¯¹é½æ¨¡å¼ä¿¡æ¯é€šå¸¸åœ¨å±å¹•ä¸Šå¯è§ã€‚å¦å¤–ï¼Œå°è¯•ä½¿ç”¨é€Ÿç‡ç±»åˆ«ï¼ˆCATæ¨¡å‹ï¼›é€šè¿‡`--p-substitution-model`ï¼‰ï¼Œå…¶ç»“æœä¸GAMMAæ¨¡å‹ä¸€æ ·å¥½ï¼Œå¹¶ä¸”é€Ÿåº¦å¿«äº†å¤§çº¦4å€ã€‚è¯·å‚é˜…[CATè®ºæ–‡](https://doi.org/10.1109/IPDPS.2006.1639535)ã€‚ CATè¿‘ä¼¼ä¹Ÿéå¸¸é€‚åˆåŒ…å«[10,000ä¸ªæˆ–æ›´å¤šç±»ç¾¤çš„æ¯”å¯¹](https://doi.org/10.1186/1471-2105-12-470)ï¼Œå¹¶ä¸”ä¸[FastTree2çš„ç±»ä¼¼CATçš„æ¨¡å‹](https://doi.org/10.1371/journal.pone.0009490)éå¸¸ç›¸ä¼¼ã€‚

## iqtree

ä¸ä¸Šé¢çš„`raxml`å’Œ`raxml-rapid-bootstrap`æ–¹æ³•ç±»ä¼¼ï¼Œæˆ‘ä»¬ä¸º[IQ-TREE](https://doi.org/10.1093/molbev/msu300)æä¾›äº†ç±»ä¼¼çš„åŠŸèƒ½ï¼š`iqtree`å’Œ`iqtree-ultrafast-bootstrap`ã€‚ ä¸`fastree`å’Œ`raxml`é€‰é¡¹ç›¸æ¯”ï¼ŒIQ-TREEçš„ç‹¬ç‰¹ä¹‹å¤„åœ¨äºå®ƒå¯ä»¥è®¿é—®286ä¸ªæ ¸è‹·é…¸å–ä»£æ¨¡å‹ã€‚ IQ-TREEè¿˜å¯ä»¥é€šè¿‡å…¶å†…ç½®çš„ModelFinderç®—æ³•åœ¨æ„å»ºæ ‘ä¹‹å‰ç¡®å®šå…¶ä¸­å“ªä¸ªæ¨¡å‹æœ€é€‚åˆæ‚¨çš„æ•°æ®é›†ã€‚ è¿™æ˜¯QIIME 2ä¸­çš„é»˜è®¤è®¾ç½®ï¼Œä½†è¯·æ”¾å¿ƒï¼Œæ‚¨å¯ä»¥é€šè¿‡--p-substitution-modelæ ‡å¿—è®¾ç½®286ä¸ª[æ ¸è‹·é…¸æ›¿æ¢æ¨¡å‹](https://doi.org/10.1016/j.dci.2004.07.007)ä¸­çš„ä»»ä½•ä¸€ä¸ªï¼Œä¾‹å¦‚ æ‚¨å¯ä»¥å°†æ¨¡å‹è®¾ç½®ä¸º`HKY+I+G`ï¼Œè€Œä¸æ˜¯é»˜è®¤çš„`MFP`ï¼ˆåŸºæœ¬çš„ç®€å†™ï¼šâ€œåœ¨ç¡®å®šç”±ModelFinderç¡®å®šçš„æœ€ä½³æ‹Ÿåˆæ¨¡å‹åå»ºç«‹ç³»ç»Ÿå‘è‚²â€ï¼‰ã€‚ è¯·è®°ä½ï¼Œé€šè¿‡ModelFinderè¿›è¡Œæ¨¡å‹æµ‹è¯•æ‰€éœ€çš„é¢å¤–è®¡ç®—æ—¶é—´ã€‚

ä½¿ç”¨é»˜è®¤è®¾ç½®å’Œè‡ªåŠ¨æ¨¡å‹é€‰æ‹©ï¼ˆ`MFP`ï¼‰è¿è¡Œ[iqtreeå‘½ä»¤](https://docs.qiime2.org/2020.2/plugins/available/phylogeny/iqtree/)çš„æœ€ç®€å•æ–¹æ³•å¦‚ä¸‹ï¼š

    # 12s
    time qiime phylogeny iqtree \
      --i-alignment masked-aligned-rep-seqs.qza \
      --o-tree iqt-tree.qza \
      --verbose

æ˜¾ç¤ºå¦‚ä¸‹ç»“æœï¼š

    Running external command line application. This may print messages to stdout and/or stderr.
    The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.
    
    Command: iqtree -st DNA --runs 1 -s /tmp/qiime2-archive-5pz537ii/0f292022-14c0-431e-83ea-87e07fa04b35/data/aligned-dna-sequences.fasta -m MFP -pre /tmp/tmpio8g5og8/q2iqtree -nt 1
    
    IQ-TREE multicore version 1.6.12 for Linux 64-bit built Aug 15 2019
    Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
    Heiko Schmidt, Dominik Schrempf, Michael Woodhams.
    
    Host:    biocloud (AVX2, FMA3, 1007 GB RAM)
    Command: iqtree -st DNA --runs 1 -s /tmp/qiime2-archive-5pz537ii/0f292022-14c0-431e-83ea-87e07fa04b35/data/aligned-dna-sequences.fasta -m MFP -pre /tmp/tmpio8g5og8/q2iqtree -nt 1
    Seed:    294079 (Using SPRNG - Scalable Parallel Random Number Generator)
    Time:    Thu May 14 16:59:37 2020
    Kernel:  AVX+FMA - 1 threads (96 CPU cores detected)
    
    HINT: Use -nt option to specify number of threads because your CPU has 96 cores!
    HINT: -nt AUTO will automatically determine the best number of threads to use.
    
    Reading alignment file /tmp/qiime2-archive-5pz537ii/0f292022-14c0-431e-83ea-87e07fa04b35/data/aligned-dna-sequences.fasta ... Fasta format detected
    Alignment most likely contains DNA/RNA sequences
    Alignment has 20 sequences with 214 columns, 157 distinct patterns
    104 parsimony-informative, 33 singleton sites, 77 constant sites
                                              Gap/Ambiguity  Composition  p-value
       1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
       2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
    ......
      20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
    ****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value<5%; df=3)
    
    
    Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.003 seconds
    NOTE: ModelFinder requires 1 MB RAM!
    ModelFinder will test 286 DNA models (sample size: 214) ...
     No. Model         -LnL         df  AIC          AICc         BIC
      1  JC            1425.567     37  2925.133     2941.111     3049.674
      2  JC+I          1424.229     38  2924.459     2941.396     3052.366
    ......
    279  GTR+F+R3      1387.357     49  2872.715     2902.593     3037.647
    Akaike Information Criterion:           TVMe+R2
    Corrected Akaike Information Criterion: TVMe+R2
    Bayesian Information Criterion:         TVMe+R2
    Best-fit model: TVMe+R2 chosen according to BIC
    
    All model information printed to /tmp/tmpio8g5og8/q2iqtree.model.gz
    CPU time for ModelFinder: 1.280 seconds (0h:0m:1s)
    Wall-clock time for ModelFinder: 1.326 seconds (0h:0m:1s)
    
    NOTE: 0 MB RAM (0 GB) is required!
    Estimate model parameters (epsilon = 0.100)
    1. Initial log-likelihood: -1416.738
    2. Current log-likelihood: -1396.508
    3. Current log-likelihood: -1389.944
    4. Current log-likelihood: -1389.313
    5. Current log-likelihood: -1389.127
    6. Current log-likelihood: -1389.021
    Optimal log-likelihood: -1388.956
    Rate parameters:  A-C: 0.16336  A-G: 2.04987  A-T: 1.55702  C-G: 0.78640  C-T: 2.04987  G-T: 1.00000
    Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
    Site proportion and rates:  (0.668,0.373) (0.332,2.263)
    Parameters optimization took 6 rounds (0.045 sec)
    Computing ML distances based on estimated model parameters... 0.004 sec
    Computing BIONJ tree...
    0.000 seconds
    Log-likelihood of BIONJ tree: -1389.065
    --------------------------------------------------------------------
    |             INITIALIZING CANDIDATE TREE SET                      |
    --------------------------------------------------------------------
    Generating 98 parsimony trees... 0.152 second
    Computing log-likelihood of 98 initial trees ... 0.188 seconds
    Current best score: -1388.956
    
    Do NNI search on 20 best initial trees
    Estimate model parameters (epsilon = 0.100)
    BETTER TREE FOUND at iteration 1: -1382.373
    Iteration 10 / LogL: -1382.379 / Time: 0h:0m:0s
    Iteration 20 / LogL: -1382.373 / Time: 0h:0m:0s
    Finish initializing candidate tree set (1)
    Current best tree score: -1382.373 / CPU time: 0.600
    Number of iterations: 20
    --------------------------------------------------------------------
    |               OPTIMIZING CANDIDATE TREE SET                      |
    --------------------------------------------------------------------
    UPDATE BEST LOG-LIKELIHOOD: -1382.372
    Iteration 30 / LogL: -1386.029 / Time: 0h:0m:0s (0h:0m:2s left)
    Iteration 40 / LogL: -1384.603 / Time: 0h:0m:1s (0h:0m:1s left)
    Iteration 50 / LogL: -1384.608 / Time: 0h:0m:1s (0h:0m:1s left)
    Estimate model parameters (epsilon = 0.100)
    BETTER TREE FOUND at iteration 52: -1382.030
    Iteration 60 / LogL: -1384.345 / Time: 0h:0m:1s (0h:0m:2s left)
    BETTER TREE FOUND at iteration 61: -1382.027
    Iteration 70 / LogL: -1382.100 / Time: 0h:0m:1s (0h:0m:2s left)
    Iteration 80 / LogL: -1382.033 / Time: 0h:0m:1s (0h:0m:1s left)
    Iteration 90 / LogL: -1382.871 / Time: 0h:0m:2s (0h:0m:1s left)
    Iteration 100 / LogL: -1382.875 / Time: 0h:0m:2s (0h:0m:1s left)
    Iteration 110 / LogL: -1382.098 / Time: 0h:0m:2s (0h:0m:1s left)
    Iteration 120 / LogL: -1382.879 / Time: 0h:0m:2s (0h:0m:0s left)
    Iteration 130 / LogL: -1382.498 / Time: 0h:0m:2s (0h:0m:0s left)
    Iteration 140 / LogL: -1382.502 / Time: 0h:0m:2s (0h:0m:0s left)
    Iteration 150 / LogL: -1382.430 / Time: 0h:0m:3s (0h:0m:0s left)
    Iteration 160 / LogL: -1382.118 / Time: 0h:0m:3s (0h:0m:0s left)
    TREE SEARCH COMPLETED AFTER 162 ITERATIONS / Time: 0h:0m:3s
    
    --------------------------------------------------------------------
    |                    FINALIZING TREE SEARCH                        |
    --------------------------------------------------------------------
    Performs final model parameters optimization
    Estimate model parameters (epsilon = 0.010)
    1. Initial log-likelihood: -1382.027
    Optimal log-likelihood: -1382.018
    Rate parameters:  A-C: 0.18868  A-G: 1.82567  A-T: 1.52062  C-G: 0.76578  C-T: 1.82567  G-T: 1.00000
    Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
    Site proportion and rates:  (0.719,0.413) (0.281,2.500)
    Parameters optimization took 1 rounds (0.005 sec)
    BEST SCORE FOUND : -1382.018
    Total tree length: 7.018
    
    Total number of iterations: 162
    CPU time used for tree search: 2.988 sec (0h:0m:2s)
    Wall-clock time used for tree search: 3.278 sec (0h:0m:3s)
    Total CPU time used: 3.052 sec (0h:0m:3s)
    Total wall-clock time used: 3.349 sec (0h:0m:3s)
    
    Analysis results written to: 
      IQ-TREE report:                /tmp/tmpio8g5og8/q2iqtree.iqtree
      Maximum-likelihood tree:       /tmp/tmpio8g5og8/q2iqtree.treefile
      Likelihood distances:          /tmp/tmpio8g5og8/q2iqtree.mldist
      Screen log file:               /tmp/tmpio8g5og8/q2iqtree.log
    
    Date and Time: Thu May 14 16:59:42 2020
    Saved Phylogeny[Unrooted] to: iqt-tree.qza

**è¾“å‡ºå¯¹è±¡ï¼š**
- `iqt-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-tree.qza)

### æŒ‡å®šæ›¿ä»£æ¨¡å‹Specifying a substitution model

æˆ‘ä»¬è¿˜å¯ä»¥è®¾ç½®æˆ‘ä»¬é€‰æ‹©çš„æ›¿ä»£æ¨¡å‹ã€‚æ‚¨å¯èƒ½åœ¨è§‚çœ‹ä¸Šä¸€ä¸ªå‘½ä»¤çš„å±å¹•è¾“å‡ºæ—¶æ³¨æ„åˆ°ï¼Œå·²æ³¨æ„åˆ°ModelFinderé€‰æ‹©çš„æœ€ä½³æ‹Ÿåˆæ¨¡å‹ã€‚ä¸ºäº†äº‰è¾©èµ·è§ï¼Œå‡è®¾æœ€ä½³é€‰æ‹©çš„æ¨¡å‹æ˜¾ç¤ºä¸º`GTR+F+I+G4`ã€‚ Fä»…æ˜¯ä¸€ç§è¡¨ç¤ºæ³•ï¼Œå®ƒä½¿æˆ‘ä»¬çŸ¥é“ï¼Œå¦‚æœç»™å®šæ¨¡å‹æ”¯æŒä¸ç›¸ç­‰çš„åŸºé¢‘ï¼Œåˆ™é»˜è®¤æƒ…å†µä¸‹å°†ä½¿ç”¨ç»éªŒåŸºé¢‘ã€‚ä½¿ç”¨ç»éªŒåŸºç¡€é¢‘ç‡ï¼ˆFï¼‰è€Œä¸æ˜¯ä¼°è®¡å®ƒä»¬ï¼Œå¯ä»¥å¤§å¤§å‡å°‘è®¡ç®—æ—¶é—´ã€‚ iqtreeæ’ä»¶åœ¨å‘½ä»¤è¡Œæä¾›çš„æ¨¡å‹ç¬¦å·å†…å°†ä¸æ¥å—Fï¼Œå› ä¸ºå¯¹äºé€‚å½“çš„æ¨¡å‹ï¼Œè¿™å§‹ç»ˆä¼šè‡ªåŠ¨éšå«ã€‚å¦å¤–ï¼Œiqtreeæ’ä»¶ä»…æ¥å—åœ¨æ¨¡å‹ç¬¦å·ä¸­æŒ‡å®šçš„Gè€Œä¸æ˜¯G4ã€‚ 4åªæ˜¯å¦ä¸€ä¸ªæ˜ç¡®çš„è¡¨ç¤ºæ³•ï¼Œä»¥æé†’æˆ‘ä»¬é»˜è®¤æƒ…å†µä¸‹é‡‡ç”¨å››ä¸ªæ¯”ç‡ç±»åˆ«ã€‚åœ¨å‘½ä»¤è¡Œä¸Šæä¾›æ¨¡å‹ç¬¦å·æ—¶ï¼Œæ’ä»¶ä½¿ç”¨çš„ç¬¦å·æ–¹æ³•ä»…æœ‰åŠ©äºä¿æŒç®€å•æ€§å’Œç†Ÿæ‚‰æ€§ã€‚å› æ­¤ï¼Œç®€å•æ¥è¯´ï¼Œæˆ‘ä»¬åªéœ€è¦è¾“å…¥`GTR+I+G`ä½œä¸ºæˆ‘ä»¬çš„è¾“å…¥æ¨¡å‹ï¼š

    # 9s
    time qiime phylogeny iqtree \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-substitution-model 'GTR+I+G' \
      --o-tree iqt-gtrig-tree.qza \
      --verbose

æ˜¾ç¤ºç»“æœåŒä¸Šé¢ç±»ä¼¼ï¼Œçœç•¥é‡å¤éƒ¨åˆ†ã€‚ 

    Date and Time: Thu May 14 17:04:04 2020
    Saved Phylogeny[Unrooted] to: iqt-gtrig-tree.qza

**è¾“å‡ºå¯¹è±¡ï¼š**
- `iqt-gtrig-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-gtrig-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-gtrig-tree.qza)

è®©æˆ‘ä»¬é‡æ–°è¿è¡Œä¸Šé¢çš„å‘½ä»¤å¹¶æ·»åŠ `--p-fast`é€‰é¡¹ã€‚ æ­¤é€‰é¡¹ä»…ä¸`iqtree`æ–¹æ³•å…¼å®¹ï¼Œç±»ä¼¼äº`fasttree`æ‰§è¡Œçš„å¿«é€Ÿæœç´¢ã€‚ ğŸï¸å…¶æ¬¡ï¼Œæˆ‘ä»¬è¿˜è¦æ‰§è¡Œå¤šä¸ªæ ‘æœç´¢ï¼Œå¹¶ä¿æŒå…¶ä¸­çš„æœ€å¥½æ ‘ï¼ˆå°±åƒæˆ‘ä»¬ä¹‹å‰ä½¿ç”¨`raxml --p-n-searches ...`å‘½ä»¤æ‰€åšçš„é‚£æ ·ï¼‰ï¼š

    # 8s
    time qiime phylogeny iqtree \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-substitution-model 'GTR+I+G' \
      --p-fast \
      --p-n-runs 10 \
      --o-tree iqt-gtrig-fast-ms-tree.qza \
      --verbose

æ˜¾ç¤ºç»“æœåŒä¸Šé¢ç±»ä¼¼ï¼Œçœç•¥é‡å¤éƒ¨åˆ†ã€‚ 

    Date and Time: Thu May 14 17:15:01 2020
    Saved Phylogeny[Unrooted] to: iqt-gtrig-fast-ms-tree.qza

**è¾“å‡ºå¯¹è±¡ï¼š**
- `iqt-gtrig-fast-ms-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-gtrig-fast-ms-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-gtrig-fast-ms-tree.qza)

### å•ææ£€éªŒ Single branch tests

IQ-TREEæä¾›ä¸€äº›å•ææ£€éªŒçš„æ–¹æ³•

1. [SH-aLRT](https://doi.org/10.1093/sysbio/syq010)ï¼š`--p-alrt [INT >= 1000]`
2. [aBayes](https://doi.org/10.1093/sysbio/syr041): `--p-abayes [TRUE | FALSE]`
3. [local bootstrap test](https://doi.org/10.1007/BF0249864): `--p-lbp [INT >= 1000]`

å•åˆ†æ”¯æ£€éªŒé€šå¸¸è¢«ç”¨ä½œæˆ‘ä»¬ä¸Šé¢è®¨è®ºè¿‡çš„è‡ªå±•æ–¹æ³•çš„æ›¿ä»£æ–¹æ³•ï¼Œå› ä¸ºå®ƒä»¬çš„é€Ÿåº¦è¦å¿«å¾—å¤šï¼Œå¹¶ä¸”åœ¨æ„å»ºå¤§å‹ç³»ç»Ÿå‘è‚²æ ‘ï¼ˆä¾‹å¦‚ï¼Œ> 10,000ä¸ªç±»ç¾¤ï¼‰æ—¶ç»å¸¸è¢«æ¨èä½¿ç”¨ã€‚ è¿™ä¸‰ç§æ–¹æ³•å‡å¯åŒæ—¶åº”ç”¨ï¼Œå¹¶åœ¨iTOLä¸­è§†ä¸ºç‹¬ç«‹çš„è‡ªå±•æ”¯æŒç‡ã€‚ è¿™äº›å€¼å§‹ç»ˆæŒ‰*alrt / lbp / abayes*çš„ä»¥ä¸‹é¡ºåºåˆ—å‡ºã€‚ æˆ‘ä»¬å°†ç»§ç»­å¹¶åœ¨ä¸‹ä¸€ä¸ªå‘½ä»¤ä¸­åº”ç”¨æ‰€æœ‰åˆ†æ”¯æ£€éªŒï¼ŒåŒæ—¶æŒ‡å®šä¸ä¸Šè¿°ç›¸åŒçš„æ›¿ä»£æ¨¡å‹ã€‚ éšæ—¶å°†å…¶ä¸`--p-fast`é€‰é¡¹ç»“åˆä½¿ç”¨ã€‚ ğŸ˜‰

    # 12s
    time qiime phylogeny iqtree \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-alrt 1000 \
      --p-abayes \
      --p-lbp 1000 \
      --p-substitution-model 'GTR+I+G' \
      --o-tree iqt-sbt-tree.qza \
      --verbose

**è¾“å‡ºå¯¹è±¡ï¼š**
- `iqt-sbt-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-sbt-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-sbt-tree.qza)

> å°è´´å£«ï¼šIQ-TREEæœç´¢è®¾ç½®ã€‚ `iqtree`æœ‰å¾ˆå¤šå¯è°ƒæ•´çš„å‚æ•°ï¼Œå¯ä»¥å¯¹å…¶è¿›è¡Œä¿®æ”¹ï¼Œä»¥æ”¹å–„é€šè¿‡â€œæ ‘ç©ºé—´â€çš„æœç´¢ï¼Œå¹¶é˜²æ­¢æœç´¢ç®—æ³•é™·å…¥å±€éƒ¨æœ€ä¼˜çŠ¶æ€ã€‚åœ¨è¿™æ–¹é¢æœ‰å¸®åŠ©çš„ä¸€ç§æœ€ä½³å®è·µæ˜¯è°ƒæ•´ä»¥ä¸‹å‚æ•°ï¼š`--p-perturb-nni-strength`å’Œ`--p-stop-iter`ï¼ˆæ¯ä¸ªå‚æ•°åˆ†åˆ«æ˜ å°„åˆ°`iqtree`çš„`-pers`å’Œ`-nstop`æ ‡å¿—ï¼‰ã€‚ç®€è€Œè¨€ä¹‹ï¼ŒNNIï¼ˆnearest-neighbor interchange, æœ€è¿‘é‚»å±…äº¤æ¢ï¼‰æ‰°åŠ¨çš„å€¼è¶Šå¤§ï¼Œâ€œæ ‘ç©ºé—´â€ä¸­çš„è·³è·ƒå°±è¶Šå¤§ã€‚è¯¥å€¼åº”è®¾ç½®å¾—è¶³å¤Ÿé«˜ï¼Œä»¥å…è®¸æœç´¢ç®—æ³•é¿å…é™·å…¥å±€éƒ¨æœ€ä¼˜çŠ¶æ€ï¼Œä½†åˆä¸èƒ½è®¾ç½®å¾—å¤ªé«˜ï¼Œä»¥å…æœç´¢å¶ç„¶åœ¨â€œæ ‘ç©ºé—´â€ä¸­è·³è·ƒã€‚ä¹Ÿå°±æ˜¯è¯´ï¼ŒåƒGoldilockså’Œä¸‰ä¸ªğŸ»ï¼Œæ‚¨éœ€è¦æ‰¾åˆ°ä¸€ä¸ªâ€œæ°å¥½â€çš„è®¾ç½®ï¼Œæˆ–è€…è‡³å°‘åœ¨ä¸€ä¸ªåˆç†çš„èŒƒå›´å†…ã€‚ä¸€ç§è¯„ä¼°æ–¹æ³•æ˜¯ä½¿ç”¨`--verbose`æ ‡å¿—è¿›è¡Œä¸€äº›çŸ­æœŸè¯•ç”¨ã€‚å¦‚æœæ‚¨çœ‹åˆ°ä¼¼ç„¶å€¼è·³å¾—å¾ˆå¤§ï¼Œåˆ™å¯ä»¥é™ä½`--p-perturb-nni-strength`çš„å€¼ã€‚è‡³äºåœæ­¢æ ‡å‡†ï¼Œå³`--p-stop-iter`ï¼Œæ­¤å€¼è¶Šé«˜ï¼Œæ‚¨åœ¨â€œæ ‘ç©ºé—´â€ä¸­çš„æœç´¢å°±è¶Šå½»åº•ã€‚è¯·æ³¨æ„ï¼Œå¢åŠ æ­¤å€¼ä¹Ÿå¯èƒ½ä¼šå¢åŠ è¿è¡Œæ—¶é—´ã€‚ä¹Ÿå°±æ˜¯è¯´ï¼Œæœç´¢å°†ç»§ç»­è¿›è¡Œï¼Œç›´åˆ°é‡‡æ ·äº†è®¸å¤šæ ‘ï¼ˆä¾‹å¦‚100ï¼ˆé»˜è®¤ï¼‰ï¼‰ï¼Œè€Œæ²¡æœ‰æ‰¾åˆ°æ›´å¥½çš„è¯„åˆ†æ ‘ã€‚å¦‚æœæ‰¾åˆ°æ›´å¥½çš„æ ‘ï¼Œåˆ™è®¡æ•°å™¨é‡ç½®ï¼Œç„¶åç»§ç»­æœç´¢ã€‚å½“ç»™å®šçš„æ•°æ®é›†åŒ…å«è®¸å¤šçŸ­åºåˆ—ï¼ˆå¯¹äºå¾®ç”Ÿç‰©ç»„è°ƒæŸ¥æ•°æ®éå¸¸æ™®éï¼‰æ—¶ï¼Œè¿™ä¸¤ä¸ªå‚æ•°å€¼å¾—ç‰¹åˆ«è€ƒè™‘ã€‚æˆ‘ä»¬å¯ä»¥å¯¹åŸå§‹å‘½ä»¤è¿›è¡Œä¿®æ”¹ï¼Œä»¥åŒ…å«è¿™äº›é¢å¤–çš„å‚æ•°ï¼Œå¹¶å»ºè®®å¯¹çŸ­åºåˆ—è¿›è¡Œä¿®æ”¹ï¼Œä¾‹å¦‚ï¼Œè¾ƒä½çš„æ‰°åŠ¨å¼ºåº¦å€¼ï¼ˆè¾ƒçŸ­çš„è¯»å–ä¸åŒ…å«å¤ªå¤šçš„ç³»ç»Ÿå‘è‚²ä¿¡æ¯ï¼Œå› æ­¤ï¼Œæˆ‘ä»¬åº”é™åˆ¶åœ¨â€œæ ‘ç©ºé—´â€ä¸­è·³è½¬çš„è·ç¦»â€ï¼‰å’Œæ›´å¤šçš„åœæ­¢è¿­ä»£ã€‚æœ‰å…³é»˜è®¤å‚æ•°è®¾ç½®çš„æ›´å¤šè¯¦ç»†ä¿¡æ¯ï¼Œè¯·å‚è§IQ-TREEå‘½ä»¤å‚è€ƒã€‚æœ€åï¼Œæˆ‘ä»¬å°†è®©iqtreeæ‰§è¡Œæ¨¡å‹æµ‹è¯•ï¼Œå¹¶è‡ªåŠ¨ç¡®å®šè¦ä½¿ç”¨çš„CPUæ ¸å¿ƒçš„æœ€ä½³æ•°é‡ã€‚

    # 11s
    time qiime phylogeny iqtree \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-perturb-nni-strength 0.2 \
      --p-stop-iter 200 \
      --p-n-cores 1 \
      --o-tree iqt-nnisi-fast-tree.qza \
      --verbose

**è¾“å‡ºå¯¹è±¡ï¼š**
- `iqt-nnisi-fast-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-fast-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-nnisi-fast-tree.qza)


### iqtree-ultrafast-bootstrap

æ ¹æ®æˆ‘ä»¬åœ¨ä¸Šé¢çš„`raxml-rapid-bootstrap`éƒ¨åˆ†ä¸­çš„è®¨è®ºï¼Œæˆ‘ä»¬è¿˜å¯ä»¥ä½¿ç”¨IQ-TREEæ¥è¯„ä¼°é€šè¿‡[è¶…å¿«é€Ÿè‡ªå±•ç®—æ³•](https://doi.org/10.1093/molbev/msx281)åœ¨ç³»ç»Ÿè¿›åŒ–ç³»ç»Ÿä¸­å¯¹æ‹†åˆ†/åˆ’åˆ†çš„æ”¯æŒç¨‹åº¦ã€‚ ä¸‹é¢ï¼Œæˆ‘ä»¬å°†åº”ç”¨æ’ä»¶çš„[ultrafast bootstrapå‘½ä»¤](https://docs.qiime2.org/2020.2/plugins/available/phylogeny/iqtree-ultrafast-bootstrap/)ï¼šè‡ªåŠ¨æ¨¡å‹é€‰æ‹©ï¼ˆ`MFP`ï¼‰ï¼Œæ‰§è¡Œ`1000`æ¬¡è¿­ä»£è‡ªå±•ï¼ˆæœ€ä½è¦æ±‚ï¼‰ï¼Œè®¾ç½®ç›¸åŒçš„å»ºè®®å‚æ•°ä»¥ä»çŸ­åºåˆ—æ„å»ºç³»ç»Ÿå‘è‚²ï¼Œå¹¶è‡ªåŠ¨ç¡®å®šæœ€ä½³CPUä½¿ç”¨æ•°é‡ï¼š

    # 15s
    time qiime phylogeny iqtree-ultrafast-bootstrap \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-perturb-nni-strength 0.2 \
      --p-stop-iter 200 \
      --p-n-cores 1 \
      --o-tree iqt-nnisi-bootstrap-tree.qza \
      --verbose

**è¾“å‡ºå¯¹è±¡ï¼š**
- `iqt-nnisi-bootstrap-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-nnisi-bootstrap-tree.qza)

### ä¸ufbootä¸€èµ·æ‰§è¡Œå•åˆ†æ”¯æ£€éªŒ Perform single branch tests alongside ufboot

æˆ‘ä»¬è¿˜å¯ä»¥å°†å•åˆ†æ”¯æ£€éªŒæ–¹æ³•ä¸è¶…å¿«é€Ÿè‡ªå±•åŒæ—¶åº”ç”¨ã€‚ æ”¯æŒå€¼å°†å§‹ç»ˆæŒ‰ä»¥ä¸‹é¡ºåºè¡¨ç¤ºï¼š*alrt / lbp / abayes / ufboot*ã€‚ åŒæ ·ï¼Œè¿™äº›å€¼å¯ä»¥è§†ä¸º[iTOL](https://itol.embl.de/)ä¸­å•ç‹¬åˆ—å‡ºçš„è‡ªå±•æ”¯æŒç‡ã€‚ æˆ‘ä»¬è¿˜å°†åƒä¹‹å‰ä¸€æ ·æŒ‡å®šä¸€ä¸ªæ¨¡å‹ã€‚

    # 16s
    time qiime phylogeny iqtree-ultrafast-bootstrap \
      --i-alignment masked-aligned-rep-seqs.qza \
      --p-perturb-nni-strength 0.2 \
      --p-stop-iter 200 \
      --p-n-cores 1 \
      --p-alrt 1000 \
      --p-abayes \
      --p-lbp 1000 \
      --p-substitution-model 'GTR+I+G' \
      --o-tree iqt-nnisi-bootstrap-sbt-gtrig-tree.qza \
      --verbose

**è¾“å‡ºå¯¹è±¡ï¼š**
- `iqt-nnisi-bootstrap-sbt-gtrig-tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-sbt-gtrig-tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-nnisi-bootstrap-sbt-gtrig-tree.qza)

> å°è´´å£«ï¼šå¦‚æœéœ€è¦å‡å°‘åœ¨[UFBootæœç´¢](https://doi.org/10.1093/molbev/msx281)è¿‡ç¨‹ä¸­å‘ç”Ÿçš„[æ½œåœ¨æ¨¡å‹è¿è§„](http://www.iqtree.org/doc/Tutorial#reducing-impact-of-severe-model-violations-with-ufboot)çš„å½±å“ï¼Œå¹¶ä¸”/æˆ–è€…åªæ˜¯æƒ³æ›´åŠ ä¸¥æ ¼ï¼Œæˆ‘ä»¬å¯ä»¥åœ¨ä»»ä½•`iqtree-ultrafast-bootstrap`ä¸­æ·»åŠ `--p-bnni`é€‰é¡¹ã€‚

## ç¡®å®šæ ‘çš„æ ¹ Root the phylogeny

ä¸ºäº†æ­£ç¡®ä½¿ç”¨UniFracç­‰å¤šæ ·æ€§æŒ‡æ ‡ï¼Œç³»ç»Ÿå‘è‚²å¿…é¡»ç¡®å®šæ ¹ã€‚ é€šå¸¸ï¼Œåœ¨ç”Ÿæ ¹æ ‘æ—¶é€‰æ‹©å¤–ç»„ã€‚ é€šå¸¸ï¼Œé»˜è®¤æƒ…å†µä¸‹ï¼Œä½¿ç”¨æœ€å¤§ä¼¼ç„¶æ€§çš„ç³»ç»Ÿå‘è‚²æ¨æ–­å·¥å…·é€šå¸¸ä¼šè¿”å›æ— æ ¹çš„æ ‘ã€‚

QIIME 2ä¸º[ä¸­ç‚¹æ ¹](https://docs.qiime2.org/2020.2/plugins/available/phylogeny/midpoint-root/)ä¸ºæˆ‘ä»¬çš„ç³»ç»Ÿå‘è‚²æä¾›äº†ä¸€ç§æ–¹æ³•ã€‚ å°†æ¥å¯èƒ½ä¼šæä¾›å…¶ä»–å®šæ ¹é€‰é¡¹ã€‚ ç°åœ¨ï¼Œæˆ‘ä»¬å°†ä»`iqtree-ultrafast-bootstrap`ç¡®å®šæˆ‘ä»¬è‡ªå±•æ ‘çš„æ ¹ï¼Œå¦‚ä¸‹æ‰€ç¤ºï¼š

    # 7s
    time qiime phylogeny midpoint-root \
      --i-tree iqt-nnisi-bootstrap-sbt-gtrig-tree.qza \
      --o-rooted-tree iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza

**è¾“å‡ºå¯¹è±¡ï¼š**
- `iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza)

> å°è´´å£«ï¼šiTOLæŸ¥çœ‹æé†’ã€‚ æˆ‘ä»¬å¯ä»¥é€šè¿‡iTOLæŸ¥çœ‹æˆ‘ä»¬çš„æ ‘åŠå…¶å…³è”çš„å¯¹é½æ–¹å¼ã€‚ æ‚¨éœ€è¦åšçš„å°±æ˜¯ä¸Šä¼ `iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza`æ ‘æ–‡ä»¶ã€‚ åœ¨æ™®é€šæ¨¡å¼ä¸‹æ˜¾ç¤ºæ ‘ã€‚ ç„¶åå°†`masked-aligned-rep-seqs.qza`æ–‡ä»¶æ‹–æ”¾åˆ°å¯è§†åŒ–æ–‡ä»¶ä¸Šã€‚ ç°åœ¨ï¼Œæ‚¨å¯ä»¥æŸ¥çœ‹æ¯”å¯¹æ—è¾¹çš„ç³»ç»Ÿå‘è‚²ã€‚



# æµç¨‹ Pipelines

åœ¨è¿™é‡Œï¼Œæˆ‘ä»¬å°†æ¦‚è¿°ç³»ç»Ÿå‘è‚²æµç¨‹ï¼š[`align-to-tree-mafft-fasttree`](https://docs.qiime2.org/2020.2/plugins/available/phylogeny/align-to-tree-mafft-fasttree/)

æµç¨‹çš„ä¸€ä¸ªä¼˜ç‚¹æ˜¯ï¼Œå®ƒä»¬å°†å¸¸ç”¨å‘½ä»¤çš„æœ‰åºé›†åˆç»„åˆä¸ºä¸€ä¸ªå‹ç¼©çš„ç®€å•å‘½ä»¤ã€‚ä¸ºäº†ä½¿è¿™äº›â€œä¾¿åˆ©â€ç®¡é“æ˜“äºä½¿ç”¨ï¼Œé€šå¸¸åªå‘ç”¨æˆ·å…¬å¼€ä¸€äº›é€‰é¡¹ã€‚ä¹Ÿå°±æ˜¯è¯´ï¼Œé€šè¿‡ç®¡é“æ‰§è¡Œçš„å¤§å¤šæ•°å‘½ä»¤é€šå¸¸éƒ½é…ç½®ä¸ºä½¿ç”¨é»˜è®¤é€‰é¡¹è®¾ç½®ã€‚ä½†æ˜¯ï¼Œæä¾›äº†å¯¹äºç”¨æˆ·æ¥è¯´è€ƒè™‘è®¾ç½®éå¸¸é‡è¦çš„é€‰é¡¹ã€‚é€šè¿‡ç»™å®šæµç¨‹å…¬å¼€çš„é€‰é¡¹åœ¨å¾ˆå¤§ç¨‹åº¦ä¸Šå–å†³äºå®ƒåœ¨åšä»€ä¹ˆã€‚æµç¨‹ä¹Ÿæ˜¯æ–°ç”¨æˆ·å…¥é—¨çš„å¥½æ–¹æ³•ï¼Œå› ä¸ºå®ƒæœ‰åŠ©äºä¸ºå»ºç«‹æ ‡å‡†æ“ä½œç¨‹åºå¥ å®šè‰¯å¥½å®è·µçš„åŸºç¡€ã€‚

è€Œä¸æ˜¯è¿è¡Œä¸‹é¢åˆ—å‡ºçš„ä¸€ä¸ªæˆ–å¤šä¸ªä»¥ä¸‹QIIME 2å‘½ä»¤ï¼š

1. `qiime alignment mafft ...`
2. `qiime alignment mask ...`
3. `qiime phylogeny fasttree ...`
4. `qiime phylogeny midpoint-root ...`

æˆ‘ä»¬å¯ä»¥åˆ©ç”¨æµç¨‹`align-tree-mafft-fasttree`ä¸€æ¬¡æ€§å®Œæˆä¸Šè¿°å››ä¸ªæ­¥éª¤ã€‚è¿™æ˜¯ä»æµç¨‹å¸®åŠ©æ–‡æ¡£ä¸­è·å¾—çš„æè¿°ï¼š

> è¯¥æµç¨‹å°†ä»ä½¿ç”¨MAFFTåˆ›å»ºåºåˆ—æ¯”å¯¹å¼€å§‹ï¼Œæ­¤åï¼Œç³»ç»Ÿä¸Šæ— ä¿¡æ¯æˆ–æ­§ä¹‰æ¯”å¯¹çš„æ‰€æœ‰æ¯”å¯¹åˆ—éƒ½å°†è¢«åˆ é™¤ï¼ˆå±è”½ï¼‰ã€‚äº§ç”Ÿçš„å±è”½å¯¹é½å°†ç”¨äºæ¨æ–­ç³»ç»Ÿå‘è‚²æ ‘ï¼Œç„¶ååœ¨å…¶ä¸­ç‚¹ç¡®å®šæ ¹ã€‚æµç¨‹æ¯ä¸ªæ­¥éª¤çš„è¾“å‡ºæ–‡ä»¶å°†è¢«ä¿å­˜ã€‚è¿™åŒ…æ‹¬æ¥è‡ª`q2-alignment`æ–¹æ³•çš„æœªå±è”½å’Œå·²å±è”½çš„MAFFTå¯¹é½æ–¹å¼ï¼Œä»¥åŠæ¥è‡ª`q2-phylogeny`æ–¹æ³•çš„æœ‰æ ¹å’Œæ— æ ¹ç³»ç»Ÿå‘è‚²ã€‚

åªéœ€è¿è¡Œä»¥ä¸‹å‘½ä»¤å³å¯å®Œæˆæ‰€æœ‰æ“ä½œï¼š

    # 6s
    time qiime phylogeny align-to-tree-mafft-fasttree \
      --i-sequences rep-seqs.qza \
      --output-dir mafft-fasttree-output

**è¾“å‡ºå¯¹è±¡ï¼š**
- `mafft-fasttree-output/masked_alignment.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Fmasked_alignment.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/mafft-fasttree-output/masked_alignment.qza)
- `mafft-fasttree-output/rooted_tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Frooted_tree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/mafft-fasttree-output/rooted_tree.qza)
- `mafft-fasttree-output/alignment.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Falignment.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/mafft-fasttree-output/alignment.qza)
- `mafft-fasttree-output/tree.qza`ï¼š[é¢„è§ˆ](https://view.qiime2.org/?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Ftree.qza) | [ä¸‹è½½](https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/mafft-fasttree-output/tree.qza)

æ­å–œä½ ï¼ æ‚¨ç°åœ¨çŸ¥é“äº†å¦‚ä½•åœ¨QIIME 2ä¸­æ„å»ºç³»ç»Ÿå‘è‚²æ ‘å•¦ï¼


## Reference

https://docs.qiime2.org/2020.2

Evan Bolyen*, Jai Ram Rideout*, Matthew R. Dillon*, Nicholas A. Bokulich*, Christian C. Abnet, Gabriel A. Al-Ghalith, Harriet Alexander, Eric J. Alm, Manimozhiyan Arumugam, Francesco Asnicar, Yang Bai, Jordan E. Bisanz, Kyle Bittinger, Asker Brejnrod, Colin J. Brislawn, C. Titus Brown, Benjamin J. Callahan, AndrÃ©s Mauricio Caraballo-RodrÃ­guez, John Chase, Emily K. Cope, Ricardo Da Silva, Christian Diener, Pieter C. Dorrestein, Gavin M. Douglas, Daniel M. Durall, Claire Duvallet, Christian F. Edwardson, Madeleine Ernst, Mehrbod Estaki, Jennifer Fouquier, Julia M. Gauglitz, Sean M. Gibbons, Deanna L. Gibson, Antonio Gonzalez, Kestrel Gorlick, Jiarong Guo, Benjamin Hillmann, Susan Holmes, Hannes Holste, Curtis Huttenhower, Gavin A. Huttley, Stefan Janssen, Alan K. Jarmusch, Lingjing Jiang, Benjamin D. Kaehler, Kyo Bin Kang, Christopher R. Keefe, Paul Keim, Scott T. Kelley, Dan Knights, Irina Koester, Tomasz Kosciolek, Jorden Kreps, Morgan G. I. Langille, Joslynn Lee, Ruth Ley, **Yong-Xin Liu**, Erikka Loftfield, Catherine Lozupone, Massoud Maher, Clarisse Marotz, Bryan D. Martin, Daniel McDonald, Lauren J. McIver, Alexey V. Melnik, Jessica L. Metcalf, Sydney C. Morgan, Jamie T. Morton, Ahmad Turan Naimey, Jose A. Navas-Molina, Louis Felix Nothias, Stephanie B. Orchanian, Talima Pearson, Samuel L. Peoples, Daniel Petras, Mary Lai Preuss, Elmar Pruesse, Lasse Buur Rasmussen, Adam Rivers, Michael S. Robeson, Patrick Rosenthal, Nicola Segata, Michael Shaffer, Arron Shiffer, Rashmi Sinha, Se Jin Song, John R. Spear, Austin D. Swafford, Luke R. Thompson, Pedro J. Torres, Pauline Trinh, Anupriya Tripathi, Peter J. Turnbaugh, Sabah Ul-Hasan, Justin J. J. van der Hooft, Fernando Vargas, Yoshiki VÃ¡zquez-Baeza, Emily Vogtmann, Max von Hippel, William Walters, Yunhu Wan, Mingxun Wang, Jonathan Warren, Kyle C. Weber, Charles H. D. Williamson, Amy D. Willis, Zhenjiang Zech Xu, Jesse R. Zaneveld, Yilong Zhang, Qiyun Zhu, Rob Knight & J. Gregory Caporaso#. Reproducible, interactive, scalable and extensible microbiome data science using QIIME 2. ***Nature Biotechnology***. 2019, 37: 852-857. doi:[10.1038/s41587-019-0209-9](https://doi.org/10.1038/s41587-019-0209-9)

## è¯‘è€…ç®€ä»‹

**åˆ˜æ°¸é‘«**ï¼Œåšå£«ã€‚2008å¹´æ¯•ä¸šäºä¸œåŒ—å†œå¤§å¾®ç”Ÿç‰©å­¦ï¼Œ2014å¹´äºä¸­ç§‘é™¢é—ä¼ å‘è‚²æ‰€è·ç”Ÿç‰©ä¿¡æ¯å­¦åšå£«ï¼Œ2016å¹´é—ä¼ å­¦åšå£«åå‡ºç«™ç•™æ‰€å·¥ä½œï¼Œä»»å®åŸºå› ç»„å­¦å®éªŒå®¤å·¥ç¨‹å¸ˆã€‚ç›®å‰ä¸»è¦ç ”ç©¶æ–¹å‘ä¸ºå¾®ç”Ÿç‰©ç»„æ•°æ®åˆ†æã€åˆ†ææ–¹æ³•å¼€å‘ä¸ä¼˜åŒ–å’Œç§‘å­¦ä¼ æ’­ï¼ŒQIIME 2é¡¹ç›®å‚ä¸äººã€‚ç›®å‰åœ¨***Scienceã€Nature Biotechnologyã€Cell Host & Microbeã€Current Opinion in Microbiology*** ç­‰æ‚å¿—å‘è¡¨è®ºæ–‡20ä½™ç¯‡ã€‚2017å¹´7æœˆåˆ›åŠâ€œå®åŸºå› ç»„â€å…¬ä¼—å·ï¼Œç›®å‰åˆ†äº«å®åŸºå› ç»„ã€æ‰©å¢å­åŸåˆ›æ–‡ç« 500ä½™ç¯‡ï¼Œä»£è¡¨åšæ–‡æœ‰[ã€Šæ‰©å¢å­å›¾è¡¨è§£è¯»ã€åˆ†ææµç¨‹å’Œç»Ÿè®¡ç»˜å›¾ä¸‰éƒ¨æ›²(21ç¯‡)ã€‹](https://mp.weixin.qq.com/s/u7PQn2ilsgmA6Ayu-oP1tw)ã€[ã€ŠNatureç»¼è¿°ï¼šæ‰‹æŠŠæ‰‹æ•™ä½ åˆ†æèŒç¾¤æ•°æ®(1.8ä¸‡å­—)ã€‹](https://mp.weixin.qq.com/s/F8Anj9djawaFEUQKkdE1lg)ã€[ã€ŠQIIME2ä¸­æ–‡æ•™ç¨‹(22ç¯‡)ã€‹](https://mp.weixin.qq.com/s/UFLNaJtFPH-eyd1bLRiPTQ)ç­‰ï¼Œå…³æ³¨äººæ•°8ä¸‡+ï¼Œç´¯è®¡é˜…è¯»1300ä¸‡+ã€‚

## çŒœä½ å–œæ¬¢

- 10000+: [èŒç¾¤åˆ†æ](https://mp.weixin.qq.com/s/F8Anj9djawaFEUQKkdE1lg)  
[å®å®ä¸çŒ«ç‹—](http://mp.weixin.qq.com/s/K3y3an-EaX8iaytmxdzHqA) [æDNAå‘Nature](http://mp.weixin.qq.com/s/lO5uiMjixJ6aYTjPX-IyaQ) [å®éªŒåˆ†æè°å¯¹ç»“æœå½±å“å¤§](http://mp.weixin.qq.com/s/cL_IAoPFfmelKMPMgltrfA)  [Cellå¾®ç”Ÿç‰©ä¸“åˆŠ](https://mp.weixin.qq.com/s/fN0gpD3bZJDXSp8x4ck-3Q) [è‚ é“æŒ‡æŒ¥å¤§è„‘](https://mp.weixin.qq.com/s/pZO20VGl3Tf_OtFIbZ-zWw)
- ç³»åˆ—æ•™ç¨‹ï¼š[å¾®ç”Ÿç‰©ç»„å…¥é—¨](http://mp.weixin.qq.com/s/sQyl5EctXFB95Oxg8YIasg) [Biostar](http://mp.weixin.qq.com/s/JL-n2nD6YL8vwuRtTVmQlQ) [å¾®ç”Ÿç‰©ç»„](http://mp.weixin.qq.com/s/li7SdZVaCEyFQF8h6MMh2A)  [å®åŸºå› ç»„](http://mp.weixin.qq.com/s/bcyvhFrNr6niqD13rQfZeg) 
- ä¸“ä¸šæŠ€èƒ½ï¼š[ç”Ÿä¿¡å®å…¸](http://mp.weixin.qq.com/s/2b3_8Vvv7McqCkEfUszW3A) [å­¦æœ¯å›¾è¡¨](http://mp.weixin.qq.com/s/SCT4oso_vI0UNIJZTaG95g) [é«˜åˆ†æ–‡ç« ](http://mp.weixin.qq.com/s/kD-x7K4hI5KMgGXikyLt0Q) [ä¸å¯æˆ–ç¼ºçš„äºº](http://mp.weixin.qq.com/s/1nf7vwyvC3oemkTq_pu87A) 
- ä¸€æ–‡è¯»æ‡‚ï¼š[å®åŸºå› ç»„](http://mp.weixin.qq.com/s/Vsm6BJgqsSvxEenIBrGVLw) [å¯„ç”Ÿè™«ç›Šå¤„](https://mp.weixin.qq.com/s/hX0K9TOLPnrZ6f8lUoSYag) [è¿›åŒ–æ ‘](https://mp.weixin.qq.com/s/GV8rU3FZdc8Y-x931k_yrQ)
- å¿…å¤‡æŠ€èƒ½ï¼š[æé—®](http://mp.weixin.qq.com/s/xCif04bqZB14Z4OvesK0SQ) [æœç´¢](http://mp.weixin.qq.com/s/wn2bqIPgT5UD-GP1qzkJFA)  [Endnote](http://mp.weixin.qq.com/s/SPblPs5ByPdb2C400kIK3w)
- æ–‡çŒ®é˜…è¯» [çƒ­å¿ƒè‚ ](http://mp.weixin.qq.com/s/1uBeAQ0utxuzTTtfUx_UXA) [SemanticScholar](https://mp.weixin.qq.com/s/gaQiUrRqLpfTXzjyfbua6A) [Geenmedical](https://mp.weixin.qq.com/s/hc8g64aHN7qv8YhVfrsuvQ)
- æ‰©å¢å­åˆ†æï¼š[å›¾è¡¨è§£è¯»](http://mp.weixin.qq.com/s/oiVHO2S1JgYrKXPDU6fH2g) [åˆ†ææµç¨‹](http://mp.weixin.qq.com/s/KrYyy3jjzAL0rQzVfV6h4A) [ç»Ÿè®¡ç»˜å›¾](http://mp.weixin.qq.com/s/6tNePiaDsPPzEBZjiCXIRg) 
- [16SåŠŸèƒ½é¢„æµ‹](http://mp.weixin.qq.com/s/sztbvfdf9wa-3HJXc_m8TQ)   [PICRUSt](https://mp.weixin.qq.com/s/LWtiwBbUCAadMZPaKKDMag)  [FAPROTAX](http://mp.weixin.qq.com/s/J8EwJD_PTDhqRaD7kXlK1A)  [Bugbase](https://mp.weixin.qq.com/s/1WdysPZWo0H6NSYiNpcMUQ) [Tax4Fun](http://mp.weixin.qq.com/s/dzsh44ue93xnAs7gTde7wg)
- åœ¨çº¿å·¥å…·ï¼š[16Sé¢„æµ‹åŸ¹å…»åŸº](http://mp.weixin.qq.com/s/YIrDqNvDX0XMazCGxhH1Lg) [ç”Ÿä¿¡ç»˜å›¾](http://mp.weixin.qq.com/s/O0QAQyfxnrXlFLw268B7lg)
- ç§‘ç ”ç»éªŒï¼š[äº‘ç¬”è®°](http://mp.weixin.qq.com/s/OnwhWlq3cTycf-W1rxgV7g)  [äº‘åä½œ](http://mp.weixin.qq.com/s/W5By9mZ5PI57_xFfZ_JXiw) [å…¬ä¼—å·](http://mp.weixin.qq.com/s/hd0sdBDAMqMJsXQs0pIjUg)
- ç¼–ç¨‹æ¨¡æ¿: [Shell](http://mp.weixin.qq.com/s/YevGR79NnBAF-xtrqL8gAA)  [R](http://mp.weixin.qq.com/s/OQiE882jM6pVwqTiIjyZ1Q) [Perl](http://mp.weixin.qq.com/s/u2ZmTo-z6cbN-L6KVLYNwg) 
- ç”Ÿç‰©ç§‘æ™®: [è‚ é“ç»†èŒ](http://mp.weixin.qq.com/s/3T768LA6MWujF4yuzK4MKQ) [äººä½“ä¸Šçš„ç”Ÿå‘½](http://mp.weixin.qq.com/s/_DUI6tOYTEq0Wu7K7iRTxw) [ç”Ÿå‘½å¤§è·ƒè¿›](http://mp.weixin.qq.com/s/O_0Il0G_v_aSwkUH_noZVA)  [ç»†èƒæš—æˆ˜](http://mp.weixin.qq.com/s/M35ebWAelDIK5Iqib06JzA) [äººä½“å¥¥ç§˜](https://mp.weixin.qq.com/s/xlCdN8il1hcutkYK-42fAQ)  

## å†™åœ¨åé¢

ä¸ºé¼“åŠ±è¯»è€…äº¤æµã€å¿«é€Ÿè§£å†³ç§‘ç ”å›°éš¾ï¼Œæˆ‘ä»¬å»ºç«‹äº†â€œå®åŸºå› ç»„â€ä¸“ä¸šè®¨è®ºç¾¤ï¼Œç›®å‰å·±æœ‰å›½å†…å¤–5000+ ä¸€çº¿ç§‘ç ”äººå‘˜åŠ å…¥ã€‚å‚ä¸è®¨è®ºï¼Œè·å¾—ä¸“ä¸šè§£ç­”ï¼Œæ¬¢è¿åˆ†äº«æ­¤æ–‡è‡³æœ‹å‹åœˆï¼Œå¹¶æ‰«ç åŠ ä¸»ç¼–å¥½å‹å¸¦ä½ å…¥ç¾¤ï¼ŒåŠ¡å¿…å¤‡æ³¨â€œå§“å-å•ä½-ç ”ç©¶æ–¹å‘-èŒç§°/å¹´çº§â€ã€‚æŠ€æœ¯é—®é¢˜å¯»æ±‚å¸®åŠ©ï¼Œé¦–å…ˆé˜…è¯»[ã€Šå¦‚ä½•ä¼˜é›…çš„æé—®ã€‹](http://mp.weixin.qq.com/s/H9gkepap0hy3NNskOkO44w)å­¦ä¹ è§£å†³é—®é¢˜æ€è·¯ï¼Œä»æœ«è§£å†³ç¾¤å†…è®¨è®ºï¼Œé—®é¢˜ä¸ç§èŠï¼Œå¸®åŠ©åŒè¡Œã€‚
![image](http://bailab.genetics.ac.cn/markdown/life/yongxinliu.jpg)

å­¦ä¹ æ‰©å¢å­ã€å®åŸºå› ç»„ç§‘ç ”æ€è·¯å’Œåˆ†æå®æˆ˜ï¼Œå…³æ³¨â€œå®åŸºå› ç»„â€
![image](http://bailab.genetics.ac.cn/markdown/life/metagenome.jpg)

![image](http://bailab.genetics.ac.cn/markdown/train/1809/201807.jpg)

ç‚¹å‡»é˜…è¯»åŸæ–‡ï¼Œè·³è½¬æœ€æ–°æ–‡ç« ç›®å½•é˜…è¯»
https://mp.weixin.qq.com/s/5jQspEvH5_4Xmart22gjMA